pkg/controllers/cliconfig/cli_config_controller.go

```diff
@@ -0,0 +1,472 @@
+				},
+				MaxRequestsInFlight:   1000,   // TODO this is a made up number
+				RequestTimeoutSeconds: 5 * 60, // 5 minutes
+			},
+			AuditConfig: configv1.AuditConfig{}, // TODO probably need this
+			KubeClientConfig: configv1.KubeClientConfig{
+				KubeConfig: "", // this should use in cluster config
+				ConnectionOverrides: configv1.ClientConnectionOverrides{
+					QPS:   400, // TODO figure out values
+					Burst: 400,
+				},
+			},
+		},
+		OAuthConfig: osinv1.OAuthConfig{
+			MasterCA:                    getMasterCA(), // we have valid serving certs provided by service-ca so we can use the service for loopback
+			MasterURL:                   fmt.Sprintf("https://%s.%s.svc", service.Name, service.Namespace),
+			MasterPublicURL:             fmt.Sprintf("https://%s", route.Spec.Host),
+			AlwaysShowProviderSelection: false,
+			GrantConfig: osinv1.GrantConfig{
+				Method:               osinv1.GrantHandlerDeny, // force denial as this field must be set per OAuth client
+				ServiceAccountMethod: osinv1.GrantHandlerPrompt,
+			},
+			SessionConfig: &osinv1.SessionConfig{
+				SessionSecretsFile:   "/var/config/system/secrets/v4-0-config-system-session/v4-0-config-system-session",
+				SessionMaxAgeSeconds: 5 * 60, // 5 minutes
+				SessionName:          "ssn",
+			},
+		},
+	}
+
+	cliConfigBytes, err := runtime.Encode(encoder, cliConfig)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "EncodeFailed",
+				Message: fmt.Sprintf("Failed to encode CLI config: %v", err),
+			},
+		}
+	}
+
+	observedConfig, err := grabPrefixedConfig(operatorConfig.Spec.ObservedConfig.Raw, configobservation.OAuthServerConfigPrefix)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "GetOAuthServerConfigFailed",
+				Message: fmt.Sprintf("Unable to get oauth-server configuration: %v", err),
+			},
+		}
+	}
+
+	completeConfigBytes, err := resourcemerge.MergePrunedProcessConfig(&osinv1.OsinServerConfig{}, nil, cliConfigBytes, observedConfig, operatorConfig.Spec.UnsupportedConfigOverrides.Raw)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "MergeConfigFailed",
+				Message: fmt.Sprintf("Failed to merge config with unsupportedConfigOverrides: %v", err),
+			},
+		}
+	}
+
+	expectedCLIConfig := getCliConfigMap(completeConfigBytes)
+
+	_, _, err = resourceapply.ApplyConfigMap(c.configMaps, recorder, expectedCLIConfig)
+	if err != nil {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "OAuthConfigDegraded",
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "ApplyFailed",
+				Message: fmt.Sprintf("Failed to apply CLI configuration %q: %v", expectedCLIConfig.Name, err),
+			},
+		}
+	}
+
+	return nil
```

Comment:
the actual logic gets covered up by construction of the conditions :/ I can try to deal with this in a follow-up